### PR TITLE
U:V:optionalLogin-namespace:mu module created

### PR DESCRIPTION
**Pull request by Simcsa**
- created new attribute module urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu
- attribute value depends on U:D:login-namespace:mu

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * Attribute value depends on login-namespace:mu attribute
+ *
+ * @author Simona Kruppova
+ */
+public class urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu extends UserVirtualAttributesModuleAbstract {
+
+	@Override
+	public Attribute getAttributeValue(PerunSessionImpl sess, User user, AttributeDefinition attributeDefinition) throws InternalErrorException {
+		try {
+			return sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, "urn:perun:user:attribute-def:def:login-namespace:mu");
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu.java]: This returns good value but in wrong attribute. You have to return it as optionalLogin-namespace:mu.
You can use Utils.copyAttributeToVirtualAttributeWithValue method to help you with it.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		}
+	}
+
+	public AttributeDefinition getAttributeDefinition() {
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_USER_ATTR_VIRT);
+		attr.setFriendlyName("optionalLogin-namespace:mu");
+		attr.setDisplayName("Optional MU login");
+		attr.setType(String.class.getName());
+		attr.setDescription("Optional Masaryk university login");
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu.java]: Change this to: "Masaryk University login (if available)".

note: Masaryk University with upper-case 'U'  ;-)

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			throw new ConsistencyErrorException(e);
+		}
+	}
+
+	public AttributeDefinition getAttributeDefinition() {
+		AttributeDefinition attr = new AttributeDefinition();
+		attr.setNamespace(AttributesManager.NS_USER_ATTR_VIRT);
+		attr.setFriendlyName("optionalLogin-namespace:mu");
+		attr.setDisplayName("Optional MU login");
```
Comment (licehammer) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_optionalLogin_namespace_mu.java]: Change to: "MU login (if available)"

**Comment by stavamichal** [comment]
I need this attribute very soon, so I will rework it in #1319 and close this one.

**Pull request closed without merging**